Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/helper/ImagePickDelegate.java

```diff
@@ -82,55 +88,86 @@ public void pick(ImagePickCallback callback, boolean longPressed) {
-                    reset();
-                }
-                if (clipboardURL != null) {
-                    HttpUrl finalClipboardURL = clipboardURL;
-                    instance(FileCache.class).downloadFile(clipboardURL.toString(), new FileCacheListener() {
-                        @Override
-                        public void onSuccess(RawFile file) {
-                            BackgroundUtils.ensureMainThread();
-
-                            showToast(R.string.image_url_get_success);
-                            Uri imageURL = Uri.parse(finalClipboardURL.toString());
-                            callback.onFilePicked(imageURL.getLastPathSegment(), new File(file.getFullPath()));
-                            reset();
-                        }
-
-                        @Override
-                        public void onFail(boolean notFound) {
-                            BackgroundUtils.ensureMainThread();
-
-                            showToast(R.string.image_url_get_failed);
-                            callback.onFilePickError(true);
-                            reset();
-                        }
-                    });
-                }
+                pickRemoteFile(callback);
             } else {
-                Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
-                intent.addCategory(Intent.CATEGORY_OPENABLE);
-                intent.setType("*/*");
-
-                if (intent.resolveActivity(activity.getPackageManager()) != null) {
-                    activity.startActivityForResult(intent, IMAGE_PICK_RESULT);
-                } else {
-                    Logger.e(TAG, "No activity found to get file with");
-                    callback.onFilePickError(false);
-                    reset();
-                }
+                pickLocalFile(callback);
             }
         }
     }
 
+    private void pickLocalFile(ImagePickCallback callback) {
+        PackageManager pm = getAppContext().getPackageManager();
+
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        intent.setType("*/*");
+
+        List<ResolveInfo> resolveInfos = pm.queryIntentActivities(intent, 0);
+        List<Intent> intents = new ArrayList<>(resolveInfos.size());
+
+        for (ResolveInfo info : resolveInfos) {
+            Intent newIntent = new Intent(Intent.ACTION_GET_CONTENT);
+            newIntent.addCategory(Intent.CATEGORY_OPENABLE);
+            newIntent.setPackage(info.activityInfo.packageName);
+            newIntent.setType("*/*");
+
+            intents.add(newIntent);
+        }
+
+        if (intents.size() == 1) {
+            activity.startActivityForResult(intents.get(0), IMAGE_PICK_RESULT);
+        } else if (intents.size() > 1) {
+            Intent chooser = Intent.createChooser(
+                    intents.remove(intents.size() - 1),
+                    activity.getString(R.string.image_pick_delegate_select_file_picker)
```

Comment:
Ya gotta stop doing this man. Just use getString directly.

Reply:
One does not simply break a habit.

Reply:
Done.